### PR TITLE
fix selector function documentation bug

### PR DESCRIPTION
**Pull request by cosmic-byte**
## Description
Error in the documentation for quilt push selector_fxn=x
see - https://docs.quiltdata.com/api-reference/package#Package.push
Screenshot 2020-02-26 at 10 59 10 AM

### CODE REVIEW

**Comment by mistercrunch** [comment]
# [Codecov](https://codecov.io/gh/quiltdata/quilt/pull/1548?src=pr&el=h1) Report
> Merging [#1548](https://codecov.io/gh/quiltdata/quilt/pull/1548?src=pr&el=desc) into [master](https://codecov.io/gh/quiltdata/quilt/commit/e2045478a18cc3735910fab1fb4fcc7b49e2a452&el=desc) will **decrease** coverage by `0.01%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/quiltdata/quilt/pull/1548?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master    #1548      +/-   ##
==========================================
- Coverage   83.33%   83.31%   -0.02%     
==========================================
  Files          27       27              
  Lines        3715     3717       +2     
==========================================
+ Hits         3096     3097       +1     
- Misses        619      620       +1     
```


| [Impacted Files](https://codecov.io/gh/quiltdata/quilt/pull/1548?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [quilt3/search\_util.py](https://codecov.io/gh/quiltdata/quilt/pull/1548/diff?src=pr&el=tree#diff-cXVpbHQzL3NlYXJjaF91dGlsLnB5) | `75.86% <0.00%> (-3.45%)` | :arrow_down: |
| [quilt3/packages.py](https://codecov.io/gh/quiltdata/quilt/pull/1548/diff?src=pr&el=tree#diff-cXVpbHQzL3BhY2thZ2VzLnB5) | `89.22% <0.00%> (+0.03%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/quiltdata/quilt/pull/1548?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/quiltdata/quilt/pull/1548?src=pr&el=footer). Last update [e204547...7c42027](https://codecov.io/gh/quiltdata/quilt/pull/1548?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: api/python/quilt3/packages.py [hunk 1]
```diff
@@ -1156,7 +1156,7 @@ def top_hash(self):
 
 
     @ApiTelemetry("package.push")
-    def push(self, name, registry=None, dest=None, message=None, selector_fn=lambda logical_key, package_entry: True):
+    def push(self, name, registry=None, dest=None, message=None, selector_fn=None):
```
Comment (akarve) [inline api/python/quilt3/packages.py]: this works in this specific case but has two problems:
1) does not work with future lambda kwargs
2) is not type consistent

can we instead print the lambda function by modifying the gendocs protocol? [here's how to print the lambda](https://stackoverflow.com/questions/30983875/how-do-you-read-a-lambda-function-as-a-string)

**Comment by cosmic-byte** [comment]
Close until a better approach is found.

File: api/python/quilt3/packages.py (hunk 1 above)
Comment (sir-sigurd) [reply api/python/quilt3/packages.py]: @akarve

For me this fix looks OK, as I think this is the first time when I see lambda being used as kwarg default.

Also I think it should look OK here too:
```python
help(quilt3.Package.push)

Help on function push in module quilt3.packages:

push(self, name, registry=None, dest=None, message=None, selector_fn=<function Package.<lambda> at 0x7f5d09f6de60>)
    Copies objects to path, then creates a new package that points to those objects.
...
```

File: api/python/quilt3/packages.py [hunk 2]
```diff
@@ -1193,6 +1193,10 @@ def push(self, name, registry=None, dest=None, message=None, selector_fn=lambda
         Returns:
             A new package that points to the copied objects.
         """
+        def default_selector_fn(arg1, arg2): return True  # Ignore Unused arguments warning
+
+        selector_fn = default_selector_fn if selector_fn is None else selector_fn
```
Comment (sir-sigurd) [inline api/python/quilt3/packages.py:1198]:
```suggestion
        selector_fn = lambda *args: True if selector_fn is None else selector_fn
```

File: docs/API Reference/Package.md [hunk 3]
```diff
@@ -393,10 +393,8 @@ __Returns__
 a PackageEntry
 
 ## __slots__
-Built-in mutable sequence.
-
-If no argument is given, the constructor creates a new empty list.
-The argument must be an iterable if specified.
+list() -> new empty list
+list(iterable) -> new list initialized from iterable's items
```
Comment (sir-sigurd) [inline docs/API Reference/Package.md:397]: Revert this change.
